mate-panel/panel.c

```diff
@@ -302,14 +302,12 @@ mate_panel_applet_move(PanelWidget *panel, GtkWidget *widget, gpointer data)
 static GtkWidget *
 panel_menu_get (PanelWidget *panel, PanelData *pd)
 {
-	if (!pd->menu) {
```

Comment:
Every time when `panel_menu_get` is called you re-create menu now... What happens with menu or menus that was created previously?